mergify_engine/merge_train.py

```diff
@@ -0,0 +1,440 @@
+
+        self.train.client.post(
+            f"/repos/{self.train.owner}/{self.train.repo}/git/refs",
+            json={
+                "ref": f"refs/heads/{branch_name}",
+                "sha": self.initial_current_base_sha,
+            },
+        )
+
+        for pull_number in self.parents + [self.pull_request_number]:
+            # TODO(sileht): if a merge fail we should update the summary of self.pull_request_number with
+            # the failure
+            self.train.client.post(
+                f"/repos/{self.train.owner}/{self.train.repo}/merges",
+                json={
+                    "base": branch_name,
+                    "head": f"refs/pull/{pull_number}/head",
+                    "commit_message": f"Merge of #{pull_number}",
+                },
+            )
+
+        try:
+            pulls_for_title = ", ".join(
+                [f"#{p}" for p in self.parents + [self.pull_request_number]]
+            )
+            title = f"merge-queue: testing {self.initial_current_base_sha[:7]}, {pulls_for_title} together"
+            body = ""
+            tmp_pull = self.train.client.post(
+                f"/repos/{self.train.owner}/{self.train.repo}/pulls",
+                json={
+                    "title": title,
+                    "body": body,
+                    "base": self.train.ref,
+                    "head": branch_name,
+                },
+            ).json()
+        except http.HTTPClientSideError as e:
+            if e.status_code == 422 and "No commits between" in e.message:
+                # FIXME(sileht): Delete created branch
+                return
+            raise
+
+        self.tmp_pull_number = tmp_pull["number"]
+
+        sub = utils.async_run(
+            subscription.Subscription.get_subscription(self.train.owner_id)
+        )
+        self.update_summaries(
+            context.Context(self.train.client, tmp_pull, sub),
+            None,
+            check_api.Conclusion.PENDING,
+            False,
+        )
+
+    def delete_pull(self):
+        tmp_pull = {
+            "number": self.tmp_pull_number,
+            "base": {
+                "user": {"login": self.train.owner},
+                "repo": {"name": self.train.repo},
+            },
+        }
+        sub = utils.async_run(
+            subscription.Subscription.get_subscription(self.train.owner_id)
+        )
+        tmp_ctxt = context.Context(self.train.client, tmp_pull, sub)
+
+        branch = f"{constants.MERGE_QUEUE_BRANCH_PREFIX}/{self.train.ref}/{self.pull_request_number}"
+        try:
+            self.train.client.delete(
+                f"/repos/{self.train.owner}/{self.train.repo}/git/refs/heads/{branch}"
+            )
+        except http.HTTPNotFound:
+            tmp_ctxt.log.warning("pull request does not exists", exc_info=True)
+
+        self.update_summaries(
+            tmp_ctxt,
+            None,
+            check_api.Conclusion.PENDING,
+            True,
+        )
```

Comment:
You should pass the sub here since `update_summaries` needs it.

Reply:
It's already passed via the tmp_ctxt.